Added profile for Emos GoSmart A65 14W

samotion2 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Device information
[Product Link](https://www.emos.sk/led-ziarovka-gosmart-a65-e27-14-w-94-w-1-400-lm-rgb-stmievatelna-wifi)
[Amazon Link](https://www.amazon.com.au/EMOS-Replacement-Brightness-Connectivity-Compatible/dp/B0CCJ9JHDK)
<!--
  Provide and links and additional information about the device you are adding in this PR.
-->

## Home Assistant Device information
LED bulb ZQW516R (c0mb70xdt4fqomid) by Tuya 
<!--
  Provide the Home Assistant device information. This can be found on the device page in HA, on the top left under `Device Info`.
  Please paste that information here.
-->

## Checklist
<!--
  Please check all the boxes when applicable.
-->

- [x] I have created a single PR per device. When you have multiple submissions please create separate PR's.
- [x] For lights - I have only included the gzipped files (*.gz), not the raw CSV files.
- [ ] For lights - I have provided a CSV file per supported color mode. Look that up in Developer Tools -> States

## Additional info
I encountered some issues with measurements in HS mode, so I’ll revisit that later. For now, I’ve focused on pushing color_temp.

This is my first pull request here, so please let me know if any changes are needed.
<!--
  Add any additional info we must know about the measurements here.
-->

bramstroker commented:
@samotion2 thanks, could you try again for the hs please?
I can only merge when the profile is having measurements for all the supported color modes of the light.
Because the power sensor will get unavailable otherwise, when a user switches to color mode instead of color temperature.

samotion2 commented:
I had some issues with measurements showing 0W. After searching through some posts here, I found out that Shelly (and other similar plugs) may not be very accurate under 2W, as they tend to cut off small loads. Following the suggestion, I connected a 60W incandescent bulb as a dummy load and waited for it to heat up. These are my measurements, but they don’t seem accurate, and it looks like the bulb’s readings are fluctuating.

I’m planning to try using an LED bulb as a dummy load next. Does anyone have suggestions on how I could improve the measurements, considering my limited options?

bramstroker commented:
@RubenKelevra did work on some new extension of measure tool which also measures you grid voltage and changes the dummy load power accordingly. https://github.com/bramstroker/homeassistant-powercalc/pull/2810
This is not yet finished and released but close too.
This should improve accuracy when using a dummy load and may also solve your issue.

Alternatively to using a dummy load you could also measure multiple of the same bulb when you have more.

samotion2 commented:
I reran the HS script with the LED bulb, and the data looks much better now. Is that good enough? What’s your opinion? I’m unsure about the values at the lower end of the brightness scale. Is there an easy way to check the idle power consumption, or should I measure it manually? (it seems to be showing 0W)

bramstroker commented:
> I reran the HS script with the LED bulb, and the data looks much better now. Is that good enough? What’s your opinion? I’m unsure about the values at the lower end of the brightness scale. Is there an easy way to check the idle power consumption, or should I measure it manually? (it seems to be showing 0W)

@RubenKelevra What do you think?

RubenKelevra commented:
@bramstroker while the quality improved, it's not ready for merge IMHO:

![pr_2844_compare](https://github.com/user-attachments/assets/514c27dd-bfcc-4da3-bc39-78f6f18dd4cb)

samotion2 commented:
So, what exactly do you suggest? Are you referring to just the outliers, or is there something else that could be improved? In the worst-case scenario, I could buy two more bulbs and proceed that way, but I’d prefer to keep that as a last resort.

bramstroker commented:
You mean this part right? Everything else seems find by me now.
Seems just the measurements on brightness 1 which are off.
This might be an issue with the light / firmware bug, as we had before where the light was turning on on much higher brightness.
@samotion2 you could try setting `MIN_BRIGHTNESS=4` in .env and run measurement session for a bit, but abort after this brightness level has been done.
Than compare the measurements of the current csv (brightness level 1), with the new one.

![398207070-514c27dd-bfcc-4da3-bc39-78f6f18dd4cb](https://github.com/user-attachments/assets/46afc0af-90a5-4fa6-8f6a-3052d37091ef)

RubenKelevra commented:
> You mean this part right? Everything else seems find by me now.

Yes, and that a 14 W light bulb just uses 2 W in RGB mode is also kinda odd.

@samotion2 can you confirm that the RGB colors, even if “white” is selected in the center of the color wheel in Home Assistant, is significantly dimmer?

Meaning, select the center of the color wheel and select the maximum brightness compared to the maximum brightness in the color temperature mode.

If this data is correct, it should be like 8 times lower in brightness.

> This might be an issue with the light / firmware bug, as we had before where the light was turning on on much higher brightness.

Sure, but that's already fixed in the master. The light is turned on with maximum brightness, waits a bit and then dimmed down for the first measurement. 

This also has the advantage of having the power supply and the LEDs heat up for a couple of seconds, so the system is stabilized.

This avoids this bug, as far as I can tell with my lights.

@samotion2 wrote:
> So, what exactly do you suggest? Are you referring to just the outliers, or is there something else that could be improved? In the worst-case scenario, I could buy two more bulbs and proceed that way, but I’d prefer to keep that as a last resort.

I'm sorry. We're just in an intermediary state of having a good improvement which this measurement would benefit from, but we're not yet happy with the quality of the code which implements it.

I think the best option is to wait a bit until @bramstroker merges the PR and remeasure again with it.

The difference is that it will take changing voltage in the grid into account, which affects the consumption of the incandescent light bulb, but not the consumption of the LED. Which is why there are artifacts from changing grid voltage in your measurement.

samotion2 commented:
I can confirm that the color mode is significantly dimmer than the color temperature mode. I would describe the RGB functionality of this bulb as more of a gimmick than a feature. I reran the measurements at brightness level 3 and found that this is the minimum required for the bulb to even turn on in this mode. I’ve merged the new values and also provided the raw file (which will be deleted in the final request). Should I manually edit a few outliers in the low brightness data? Is this state satisfactory, or should I wait for the new script?

RubenKelevra commented:
@samotion2 wrote:
> I reran the measurements at brightness level 3 and found that this is the minimum required for the bulb to even turn on in this mode.

Yeah, that's a common issue, that's why we don't do this anymore. We now turn them on with maximum brightness and then dim them down to 1.

So in the beginning of the measure script, it prints out, that it's turning on the light with maximum brightness, and then starts to measure after a couple of seconds.

If the light turns off, during those first measurement messages, that's an issue. Otherwise everything is fine.

You can also check this manually if you want to:

Go in HA's to the development tools and run the following actions:

```
action: light.turn_on
data:
  brightness: 256
target:
  entity_id: light.my_light_id
``` 


followed by:


```
action: light.turn_on
data:
  brightness: 1
target:
  entity_id: light.my_light_id
``` 

@samotion2 wrote:
> or should I wait for the new script?

No wait necessary, if you want to you can now checkout the [PR branch](https://github.com/bramstroker/homeassistant-powercalc/pull/2810) and run the new version. I just finished fixing all blockers. Rest is cosmetic.

LMK if you need help with that.

samotion2 commented:
I’ve also adjusted the standby power and corrected the outlier. Thanks for the tip—I tried that manually in Home Assistant, and yes, the minimum brightness for color is 3. To be honest, if this setup looks good to you, I’d like to proceed with this export. I plan to add another light when I have time, and then I can switch to using the new script. Please let me know if any additional changes are needed.

RubenKelevra reviewed (approved):
LGTM

bramstroker commented:
Thanks @samotion2 and @RubenKelevra
